File: common/valid-anagram.go
package main

import (
	"fmt"
)


func check(e error) {
	if e != nil {
		panic(e)
	}
}

func anagram(s1 string, s2 string) bool {
	if len(s1) != len(s2) {
		return false;
	}
	s1LetterCount := make(map[rune]int)
	s2LetterCount := make(map[rune]int)

	for _, ch:= range s1 {
		s1LetterCount[ch]++
	}
	for _, ch :=range s2 {
		s2LetterCount[ch]++
	}

	for key, val := range s1LetterCount {
		if s2LetterCount[key] != val {
			return false
		}
	}
	return true;
}

func main() {
	s1:= "anagram"
	s2:= "nagaram"
	fmt.Println(anagram(s1, s2))
}

File: Common Coding Problems/valid-anagram.py
def anagram(s1, s2):
    if(len(s1) != len(s2)):
        return False
    s1LetterCount = {}
    s2LetterCount = {}
    for ch in s1:
        if ch in s1LetterCount:
            s1LetterCount[ch] += 1
        else:
            s1LetterCount[ch] = 1
    for ch in s2:
        if ch in s2LetterCount:
            s2LetterCount[ch] +=1
        else:
            s2LetterCount[ch] = 1
    for key in s1LetterCount:
        if key not in s2LetterCount or s1LetterCount[key] != s2LetterCount[key]:
            return False
        return True

s1 = 'nameleses';
s2 = 'salesman'
print(anagram(s1,s2))